row_store.go

```diff
@@ -165,6 +162,28 @@ func (t *table) openRowStore(opts *rowStoreOptions) (*rowStore, wal.Offset, erro
 	return rs, walOffset, nil
 }
 
+func readWALOffset(filename string) (wal.Offset, bool, error) {
```

Comment:
This file was refactored to make the necessary pieces available to FilterAndMerge for zenotool. Logic should be basically unchanged.